app/controllers/rails_admin/main_controller.rb

```diff
@@ -44,7 +44,7 @@ def get_layout
     end
 
     def back_or_index
-      params[:return_to].presence && params[:return_to].include?(request.host) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
+      params[:return_to].presence && params[:return_to].starts_with?(request.base_url) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
```

Comment:
`request.base_url` includes both the host and protocol. Using `request.host` will cause two integration tests to fail.